presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -858,7 +858,11 @@ protected Type visitFunctionCall(FunctionCall node, StackableAstVisitorContext<C
                                         innerExpressionAnalyzer.setExpressionType(argument, getExpressionType(argument));
                                     }
                                 }
-                                return innerExpressionAnalyzer.analyze(expression, baseScope, context.getContext().expectingLambda(types)).getTypeSignature();
+                                Type type = innerExpressionAnalyzer.analyze(expression, baseScope, context.getContext().expectingLambda(types));
+                                if (expression instanceof LambdaExpression) {
```

Comment:
This logic is more intuitive isn't it?

Reply:
Curious if this `expression instanceof LambdaExpression` is required? -- since `BindExpression` is used to support Lambda capture, so if it's a `BindExpression`, it means it also has a `LambdaExpression` inside.

Would there be some corner cases that we don't want to do this `verifyNoAggregateWindowOrGroupingFunctions` check unconditionally? :)

Reply:
This is moved from `visitLambdaExpression` so the if is meant to make it logically equivalent as before. I guess it doesn't hurt to check `BindExpression` but `BindExpression` don't have a `getBody` so you'd need to handle the two separately anyways.